include/oneapi/dpl/internal/dynamic_selection_traits.h

```diff
@@ -86,7 +86,7 @@ namespace internal {
 
  } //namespace internal
 
-  constexpr int deferred_initialization = -1;
+  struct deferred_initialization{};
```

Comment:
It would be better to have a struct of perhaps deferred_initialization_t and then an inline constexpr object of that type. This would remove the need for users to construct the object when passing to a constructor, they can just refer to the inline constexpr object instead.  This would be similar to what is done in dynamic_selection_traits.h for execution_info. In that file there is a "struct task_time_t {}" and then an "inline constexpr task_time_t task_time".  When calling report, the user can then just pass task_time and doesn't need to construct an instance of task_time_t.   So you can add the inline constexpr object here and then update all of the uses in the tests where policies are created in the deferred state by passing the object instead of constructing a new instance each time.

Reply:
Done